tools/hlint.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+usage() { echo "Usage: $0 -f [all, changeset] -m [check, inplace]" 1>&2; exit 1; }
+
+files=''
+check=true
+
+while getopts ':f:m:' opt
+ do
+     case $opt in
+         f) f=${OPTARG}
+            if [ "$f" = "all" ]; then
+              files=$(find libs/ services/ -not -path "*/test/*" | grep \.hs\$)
+            elif [ "$f" = "changeset" ]; then
+              files=$(git diff --name-only HEAD~ | grep \.hs\$)
+            else
+              usage
+            fi
+            ;;
+         m) m=${OPTARG}
+            if [ "$m" = "inplace" ]; then
+              check=false
+            elif [ "$m" = "check" ]; then
+              check=true
+            else
+              usage
+            fi
+            ;;
+         *) usage;;
+     esac
+done
+
+if [ -z "${f}" ] || [ -z "${m}" ]; then
+    usage
+fi
+
+count=$(echo "$files" | wc -l | xargs echo)
```

Comment:
* Why `xargs echo` at the end?
* This is count too high if we have a filename containing a NL.
* This will not match the number of loop iterations below if we have a filename contains SP or TAB.